rest/route/annotations.go

```diff
@@ -203,41 +203,31 @@ func (h *annotationByTaskGetHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(res)
 }
 
-////////////////////////////////////////////////////////////////////////
-//
-// PUT /rest/v2/tasks/{task_id}/annotation
-
-type annotationByTaskPutHandler struct {
+// Parsing logic for task annotation put and patch routes.
+type annotationByTaskPutOrPatchHandler struct {
 	taskId string
 
 	user       gimlet.User
 	annotation *restModel.APITaskAnnotation
 }
 
-func makePutAnnotationsByTask() gimlet.RouteHandler {
-	return &annotationByTaskPutHandler{}
-}
-
-func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
-	return &annotationByTaskPutHandler{}
-}
-
-func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
+func annotationByTaskPutOrPatchParser(ctx context.Context, r *http.Request) (*annotationByTaskPutOrPatchHandler, error) {
```

Comment:
Rather than this returning an entirely new kind of handler, ti would be preferable if this function just returned the things needed by the handlers (taskId and annotation), or returned a private struct of annotationRouteParams

Reply:
done